Redeploying car file with a new connector still points to an older connector version

### Description

When we upgrade the Amazon S3 connector version from a previous version to the latest version, a null pointer is observed after deploying the latest car file.

```
ERROR_MESSAGE = Error occured in the mediation of the class mediator, ERROR_DETAIL = org.apache.synapse.SynapseException: Error occured in the mediation of the class mediator
    at org.apache.synapse.mediators.ext.ClassMediator.mediate(ClassMediator.java:100)
    at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:115)
    at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:77)
    at org.apache.synapse.mediators.template.TemplateMediator.mediate(TemplateMediator.java:147)
    at org.apache.synapse.mediators.template.InvokeMediator.mediate(InvokeMediator.java:180)
 ...
Caused by: java.lang.NullPointerException
    at software.amazon.awssdk.core.runtime.transform.StreamingRequestMarshaller.marshall(StreamingRequestMarshaller.java:48)
    at software.amazon.awssdk.core.internal.handler.BaseClientHandler.lambda$finalizeSdkHttpFullRequest$0(BaseClientHandler.java:73)
    at software.amazon.awssdk.core.internal.util.MetricUtils.measureDuration(MetricUtils.java:50)
```

According to the observation, although the latest S3 connector version was deployed, the older connector was never removed. As a result, the runtime continued executing the operation using the older connector version, which does not support the enableStreaming parameter.

Based on this, it seems that even if we use the new mediation along with the upgraded connector, the MI server still has a reference to the previous connector in the /tmp directory, and the latest mediation uses this old reference with the new mediation.



### Steps to Reproduce

- Start the MI pack by configuring required liberties to use amazons3 connector.
- Deploy a car file with previous connector version along with the mediation.
- Update the connector version to the latest version and deploy the car file without restart the server.
- Invoke the API and check the logs


### Version

4.1.0

### Environment Details (with versions)

_No response_

--- a/src/main/java/org/wso2/carbon/connector/amazons3/connection/S3ConnectionHandler.java
+++ b/src/main/java/org/wso2/carbon/connector/amazons3/connection/S3ConnectionHandler.java
@@ -54,7 +54,13 @@ public class S3ConnectionHandler implements Connection {
     }
 
     @Override
-    public void close() {}
+    public void close() {
+        if (s3Client != null) {
+            log.debug("Closing the S3 client");
+            s3Client.close();
+            s3Client = null;
+        }
+    }
 
     private S3Client createS3Client() {
 

--- a/src/main/java/org/wso2/carbon/connector/amazons3/operations/S3Config.java
+++ b/src/main/java/org/wso2/carbon/connector/amazons3/operations/S3Config.java
@@ -37,7 +37,7 @@ public class S3Config extends AbstractConnector implements ManagedLifecycle {
             ConnectionHandler handler = ConnectionHandler.getConnectionHandler();
             if (!handler.checkIfConnectionExists(connectorName, connectionName)) {
                 S3ConnectionHandler s3ConnectionHandler = new S3ConnectionHandler(configuration);
-                handler.createConnection(S3Constants.CONNECTOR_NAME, connectionName, s3ConnectionHandler);
+                handler.createConnection(S3Constants.CONNECTOR_NAME, connectionName, s3ConnectionHandler, messageContext);
             } else {
                 S3ConnectionHandler connectionHandler = (S3ConnectionHandler) handler
                         .getConnection(connectorName, connectionName);

